### DISCUSSION TITLE
Start index/buffer of firstItemIndex when prepending items

### DISCUSSION THREAD

**Discussion by Zerebokep** [petyosi/react-virtuoso#1032] [Q&A] [answered]
First off, this package is awesome and saved us much time, we'll definitely donate something after we have our first profit with our project.

Now to my question: In the prepending items example a start index of 10000 is set, we are currently doing the same as we don't know the total count of items for our use case, but on some of our pages we'll definitely have way more than 10000.

Purely in terms of performance, should we just set a big enough start index or would it be better to build something around it which increases the index dynamically?

**Comment by petyosi** [comment, marked as answer]
Any big number would work, really. The browsers (Chrome) a maximum limit of ~13M pixels for maximum element scroll height, so you should be safe.

**Reply by Zerebokep** [reply]
Okay, great, but when I put e.g. 1 million as start index there will be an array created with 1 million array entries under the hood, or does virtuoso uses a Map?

**Reply by petyosi** [reply]
No, it's just used as an offset internally. No million undefined item arrays :)
